snuba/migrations/migrate.py

```diff
@@ -36,8 +36,10 @@ def _run_schema(conn: Client, schema: Schema) -> None:
 
 def run(conn: Client, dataset: Dataset) -> None:
     schemas: MutableSequence[Schema] = []
-    writer = dataset.get_table_writer()
-    if writer:
+
+    writable_storage = dataset.get_writable_storage()
+    if writable_storage:
+        writer = writable_storage.get_table_writer()
         schemas.append(writer.get_schema())
     for storage in dataset.get_all_storages():
         schemas.append(storage.get_schemas().get_read_schema())
```

Comment:
This behavior already existed, but it seems like the storage returned by `get_writable_storage` should be in `get_all_storages` — is this not the case? Or is this attempting to work around some ordering requirement or something?

Reply:
I can't think of a case where the writable storage shouldn't just be in the list of storages (I guess it needs to be the first item in the list of storages to enforce the correct ordering). Let me come back to this later though since it's not directly related to these changes.

Reply:
Sounds good, thanks. 👍